File: db4on/tutorial/db4o-tutorial-chapters/src/com/db4o/f1/chapter6/LocalizedItemList.cs
using System.Globalization;

namespace com.db4o.f1.chapter6
{
    /// <summary>
    /// A CultureInfo aware list of objects.
    /// CultureInfo objects hold a native pointer to 
    /// a system structure.
    /// </summary>
    public class LocalizedItemList
    {
        CultureInfo _culture;
        string[] _items;

        public LocalizedItemList(CultureInfo culture, string[] items)
        {
            _culture = culture;
            _items = items;
        }

        override public string ToString()
        {
            return string.Join(string.Concat(_culture.TextInfo.ListSeparator,  " "), _items);
        }
    }
}

File: db4on/tutorial/vb/db4o-tutorial-chapters/src/com/db4o/f1/chapter6/LocalizedItemList.vb
Imports System.Globalization
Namespace com.db4o.f1.chapter6
	''' <summary>
	''' A CultureInfo aware list of objects.
	''' CultureInfo objects hold a native pointer to 
	''' a system structure.
	''' </summary>
	Public Class LocalizedItemList
		Private _culture As CultureInfo

		Private _items As String()

		Public Sub New(ByVal culture As CultureInfo, ByVal items As String())
			_culture = culture
			_items = items
		End Sub 

		Public Overloads Overrides Function ToString() As String
			Return String.Join(String.Concat(_culture.TextInfo.ListSeparator, " "), _items)
		End Function

	End Class
End Namespace
